apiserver/common/networkingcommon/networkconfigapi.go

```diff
@@ -43,25 +50,22 @@ type NetworkConfigAPI struct {
 
 // NewNetworkConfigAPI constructs a new common network configuration API
 // and returns its reference.
-func NewNetworkConfigAPI(ctx context.Context, st *state.State, cloudService common.CloudService, networkService NetworkService, getCanModify common.GetAuthFunc) (*NetworkConfigAPI, error) {
-	// TODO (manadart 2020-08-11): This is a second access of the model when
-	// being instantiated by the provisioner API.
-	// We should ameliorate repeat model access at some point,
-	// as it queries state each time.
-	mod, err := st.Model()
-	if err != nil {
-		return nil, errors.Trace(err)
-	}
-
-	cloud, err := cloudService.Cloud(ctx, mod.CloudName())
+func NewNetworkConfigAPI(
+	ctx context.Context,
+	st *state.State,
+	modelInfoService ModelInfoService,
+	networkService NetworkService,
+	getCanModify common.GetAuthFunc,
+) (*NetworkConfigAPI, error) {
+	cloudType, err := modelInfoService.GetModelCloudType(ctx)
 	if err != nil {
 		return nil, errors.Trace(err)
```

Comment:
I would suggest that this should probably become `internalerrors.Errorf("getting model cloud type: %w", err)`